docs/sql/index.md

```diff
@@ -1,407 +1,391 @@
+The `SELECT` clause determines the table that is being subscribed to.
+A subscription query may only return rows from a single table,
+and it must return the entire row.
+Individual column projections are not allowed.
 
-| SQL                                             | SATS     | Example | Notes                |
-| ----------------------------------------------- | -------- | ------- | -------------------- |
-| `CHAR`, `VARCHAR`, `NVARCHAR`, `TEXT`, `STRING` | `String` | 'hello' | Always UTF-8 encoded |
+A wildcard projection `*` is allowed when the table is unambiguous,
+but when a query contains multiple table references,
+a qualified wildcard projection `.*` is necessary.
 
-> SpacetimeDB SQL currently does not support length contraints like `CHAR(10)`.
+### FROM
 
-## Syntax
+```ebnf
+FROM table [ [AS] alias ] [ [INNER] JOIN table [ [AS] alias ] ON predicate ]
+```
 
-### Comments
+While you can only subscribe to rows from a single table,
+you may reference two tables in the `FROM` clause using a `JOIN`.
+A `JOIN` selects all combinations of rows from its input tables,
+and `ON` determines which combinations are considered.
+
+> NOTE: Joining more than two tables will be supported in the future.
+
+> IMPORTANT: If a `JOIN` is used,
+> an index **must** be defined on the join columns of both tables,
+> so that it can be evaluated efficiently.
+
+### WHERE
+
+```ebnf
+predicate
+    = expr
+    | predicate AND predicate
+    | predicate OR  predicate
+    ;
+
+expr
+    = literal
+    | column
+    | expr op expr
+    ;
+
+op
+    = '='
+    | '<'
+    | '>'
+    | '<' '='
+    | '>' '='
+    | '!' '='
+    | '<' '>'
+    ;
+
+literal
+    = INTEGER
+    | STRING
+    | HEX
+    | TRUE
+    | FALSE
+    ;
+```
 
-SQL line comments begin with `--`.
+While the `SELECT` clause determines the table,
+the `WHERE` clause determines the rows in the subscription.
 
 ```sql
--- This is a comment
+-- Which items in my inventory have been purchased and are more than $X?
+SELECT item
+FROM Inventory item
+JOIN Orders order
+ON item.item_id = order.item_id
+WHERE item.price > {X}
+
+-- Which items in my inventory have been purchased and are less than $Y?
+SELECT item
+FROM Inventory item
+JOIN Orders order
+ON item.item_id = order.item_id
+WHERE item.price < {Y}
 ```
 
-### Expressions
-
-We can express different, composable, values that are universally called `expressions`.
-
-An expression is one of the following:
-
-#### Literals
+> NOTE: Arithmetic expressions will be supported in the future.
 
-| Example   | Description |
-| --------- | ----------- |
-| `1`       | An integer. |
-| `1.0`     | A float.    |
-| `'hello'` | A string.   |
-| `true`    | A boolean.  |
+## Query and DML
 
-#### Binary operators
+### Statements
 
-| Example | Description         |
-| ------- | ------------------- |
-| `1 > 2` | Integer comparison. |
-| `1 + 2` | Integer addition.   |
+- [SELECT](#select)
```

Comment:
I don't think this link goes where you want it to; I believe it will find the first `#select` anchor, which is the above one about subscriptions, not the below one about one-off queries.